### PR TITLE
NXOS Commit Integration tests to Ansible (part 2)

### PR DESCRIPTION
**Pull request by rahushen**
##### SUMMARY

Added nxos_aaa_server, nxos_aaa_server_host, nxos_gir, nxos_gir_profile_management ITs.

##### ISSUE TYPE

 - IT Pull Request

##### ANSIBLE VERSION

```
ansible 2.4.0 (commit_it_rahushen2 b2cb6315fb) last updated 2017/09/01 16:01:25 (GMT -400)
  config file = None
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /root/agents-ci/ansible/lib/ansible
  executable location = /root/agents-ci/ansible/bin/ansible
  python version = 2.7.6 (default, Oct 26 2016, 20:30:19) [GCC 4.8.4]
```

### CODE REVIEW

File: test/integration/targets/nxos_gir/tests/common/sanity.yaml
Comment (rahushen) [inline test/integration/targets/nxos_gir/tests/common/sanity.yaml:4]: This file is fully commented out as the  module craters the device. Will uncomment when the module is fixed.

**Comment by gundalow** [comment]
Got a few issues with this when running against `nxos01.eng`

```
Software
  loader:    version N/A
  kickstart: version 7.3(0)D1(1)
  system:    version 7.3(0)D1(1)
  kickstart image file is: bootflash:///titanium-d1-kickstart.7.3.0.D1.1.bin
  kickstart compile time:  1/11/2016 16:00:00 [02/11/2016 10:30:12]
  system image file is:    bootflash:///titanium-d1.7.3.0.D1.1.bin
  system compile time:     1/11/2016 16:00:00 [02/11/2016 13:08:11]


Hardware
  cisco NX-OSv Chassis ("NX-OSv Supervisor Module")
  QEMU Virtual CPU version 2.0 with 3064860 kB of memory.
  Processor Board ID TM3E6AB9DAB

  Device name: switch
  bootflash:    3184776 kB

```



**nxos_aaa_server_host/tests/common/radius.yaml**
```
TASK [nxos_aaa_server_host : Configure radius server defaults] ***********************************************************************************************************************************************************************************
task path: /home/johnb/git/ansible-inc/nxos-prs/pr28939/test/integration/targets/nxos_aaa_server_host/tests/common/radius.yaml:13
<nxos01> connection transport is cli
<nxos01> using connection plugin network_cli
<nxos01> socket_path: /home/johnb/.ansible/pc/7b6586159a
Using module file /home/johnb/git/ansible-inc/ansible-workspace-4/lib/ansible/modules/network/nxos/nxos_aaa_server_host.py
<nxos01> ESTABLISH LOCAL CONNECTION FOR USER: johnb
<nxos01> EXEC /bin/sh -c 'echo ~ && sleep 0'
<nxos01> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/johnb/.ansible/tmp/ansible-tmp-1505310427.8-174722019319878 `" && echo ansible-tmp-1505310427.8-174722019319878="` echo /home/johnb/.ansible/tmp/ansible-tmp-1505310427.8-174722019319878 `" ) && sleep 0'
<nxos01> PUT /tmp/tmpJMWIYF TO /home/johnb/.ansible/tmp/ansible-tmp-1505310427.8-174722019319878/nxos_aaa_server_host.py
<nxos01> EXEC /bin/sh -c 'chmod u+x /home/johnb/.ansible/tmp/ansible-tmp-1505310427.8-174722019319878/ /home/johnb/.ansible/tmp/ansible-tmp-1505310427.8-174722019319878/nxos_aaa_server_host.py && sleep 0'
<nxos01> EXEC /bin/sh -c 'python /home/johnb/.ansible/tmp/ansible-tmp-1505310427.8-174722019319878/nxos_aaa_server_host.py; rm -rf "/home/johnb/.ansible/tmp/ansible-tmp-1505310427.8-174722019319878/" > /dev/null 2>&1 && sleep 0'
changed: [nxos01] => {
    "changed": true, 
    "end_state": {}, 
    "existing": {}, 
    "failed": false, 
    "invocation": {
        "module_args": {
            "acct_port": null, 
            "address": "8.8.8.8", 
            "auth_port": null, 
            "encrypt_type": null, 
            "host": null, 
            "host_timeout": null, 
            "key": null, 
            "password": null, 
            "port": null, 
            "provider": {
                "host": null, 
                "password": null, 
                "port": null, 
                "ssh_keyfile": null, 
                "timeout": null, 
                "transport": "cli", 
                "use_ssl": null, 
                "username": null, 
                "validate_certs": null
            }, 
            "server_type": "radius", 
            "ssh_keyfile": null, 
            "state": "present", 
            "tacacs_port": null, 
            "timeout": null, 
            "transport": "cli", 
            "use_ssl": null, 
            "username": null, 
            "validate_certs": null
        }
    }, 
    "proposed": {
        "address": "8.8.8.8", 
        "server_type": "radius"
    }, 
    "updates": [
        "radius-server host 8.8.8.8"
    ]
}

TASK [nxos_aaa_server_host : assert] *************************************************************************************************************************************************************************************************************
task path: /home/johnb/git/ansible-inc/nxos-prs/pr28939/test/integration/targets/nxos_aaa_server_host/tests/common/radius.yaml:21
ok: [nxos01] => {
    "changed": false, 
    "failed": false, 
    "msg": "All assertions passed"
}

TASK [nxos_aaa_server_host : Check Idempotence] **************************************************************************************************************************************************************************************************
task path: /home/johnb/git/ansible-inc/nxos-prs/pr28939/test/integration/targets/nxos_aaa_server_host/tests/common/radius.yaml:25
<nxos01> connection transport is cli
<nxos01> using connection plugin network_cli
<nxos01> socket_path: /home/johnb/.ansible/pc/7b6586159a
Using module file /home/johnb/git/ansible-inc/ansible-workspace-4/lib/ansible/modules/network/nxos/nxos_aaa_server_host.py
<nxos01> ESTABLISH LOCAL CONNECTION FOR USER: johnb
<nxos01> EXEC /bin/sh -c 'echo ~ && sleep 0'
<nxos01> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/johnb/.ansible/tmp/ansible-tmp-1505310433.68-80649056757156 `" && echo ansible-tmp-1505310433.68-80649056757156="` echo /home/johnb/.ansible/tmp/ansible-tmp-1505310433.68-80649056757156 `" ) && sleep 0'
<nxos01> PUT /tmp/tmp2fkXfo TO /home/johnb/.ansible/tmp/ansible-tmp-1505310433.68-80649056757156/nxos_aaa_server_host.py
<nxos01> EXEC /bin/sh -c 'chmod u+x /home/johnb/.ansible/tmp/ansible-tmp-1505310433.68-80649056757156/ /home/johnb/.ansible/tmp/ansible-tmp-1505310433.68-80649056757156/nxos_aaa_server_host.py && sleep 0'
<nxos01> EXEC /bin/sh -c 'python /home/johnb/.ansible/tmp/ansible-tmp-1505310433.68-80649056757156/nxos_aaa_server_host.py; rm -rf "/home/johnb/.ansible/tmp/ansible-tmp-1505310433.68-80649056757156/" > /dev/null 2>&1 && sleep 0'
changed: [nxos01] => {
    "changed": true, 
    "end_state": {}, 
    "existing": {}, 
    "failed": false, 
    "invocation": {
        "module_args": {
            "acct_port": null, 
            "address": "8.8.8.8", 
            "auth_port": null, 
            "encrypt_type": null, 
            "host": null, 
            "host_timeout": null, 
            "key": null, 
            "password": null, 
            "port": null, 
            "provider": {
                "host": null, 
                "password": null, 
                "port": null, 
                "ssh_keyfile": null, 
                "timeout": null, 
                "transport": "cli", 
                "use_ssl": null, 
                "username": null, 
                "validate_certs": null
            }, 
            "server_type": "radius", 
            "ssh_keyfile": null, 
            "state": "present", 
            "tacacs_port": null, 
            "timeout": null, 
            "transport": "cli", 
            "use_ssl": null, 
            "username": null, 
            "validate_certs": null
        }
    }, 
    "proposed": {
        "address": "8.8.8.8", 
        "server_type": "radius"
    }, 
    "updates": [
        "radius-server host 8.8.8.8"
    ]
}

TASK [nxos_aaa_server_host : assert] *************************************************************************************************************************************************************************************************************
task path: /home/johnb/git/ansible-inc/nxos-prs/pr28939/test/integration/targets/nxos_aaa_server_host/tests/common/radius.yaml:29
fatal: [nxos01]: FAILED! => {
    "assertion": "result.changed == false", 
    "changed": false, 
    "evaluated_to": false, 
    "failed": true
}

```


**nxos_aaa_server_host/tests/common/tacacs.yaml**

Pipe is not allowed for this type of message

```

TASK [nxos_aaa_server_host : debug] **************************************************************************************************************************************************************************************************************
task path: /home/johnb/git/ansible-inc/nxos-prs/pr28939/test/integration/targets/nxos_aaa_server_host/tests/common/tacacs.yaml:124
ok: [nxos01] => {
    "msg": "TRANSPORT:nxapi nxos_aaa_server_host failure detected"
}

TASK [nxos_aaa_server_host : Remove tacacs server configuration] *********************************************************************************************************************************************************************************
task path: /home/johnb/git/ansible-inc/nxos-prs/pr28939/test/integration/targets/nxos_aaa_server_host/tests/common/tacacs.yaml:128
<nxos01> connection transport is nxapi
Using module file /home/johnb/git/ansible-inc/ansible-workspace-4/lib/ansible/modules/network/nxos/nxos_aaa_server_host.py
<nxos01> ESTABLISH LOCAL CONNECTION FOR USER: johnb
<nxos01> EXEC /bin/sh -c 'echo ~ && sleep 0'
<nxos01> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/johnb/.ansible/tmp/ansible-tmp-1505310455.28-27275539476854 `" && echo ansible-tmp-1505310455.28-27275539476854="` echo /home/johnb/.ansible/tmp/ansible-tmp-1505310455.28-27275539476854 `" ) && sleep 0'
<nxos01> PUT /tmp/tmpTWXCo2 TO /home/johnb/.ansible/tmp/ansible-tmp-1505310455.28-27275539476854/nxos_aaa_server_host.py
<nxos01> EXEC /bin/sh -c 'chmod u+x /home/johnb/.ansible/tmp/ansible-tmp-1505310455.28-27275539476854/ /home/johnb/.ansible/tmp/ansible-tmp-1505310455.28-27275539476854/nxos_aaa_server_host.py && sleep 0'
<nxos01> EXEC /bin/sh -c 'python /home/johnb/.ansible/tmp/ansible-tmp-1505310455.28-27275539476854/nxos_aaa_server_host.py; rm -rf "/home/johnb/.ansible/tmp/ansible-tmp-1505310455.28-27275539476854/" > /dev/null 2>&1 && sleep 0'
fatal: [nxos01]: FAILED! => {
    "changed": false, 
    "clierror": "", 
    "code": "501", 
    "failed": true, 
    "input": "show run | inc tacacs-server.host.8.8.8.8", 
    "invocation": {
        "module_args": {
            "acct_port": null, 
            "address": "8.8.8.8", 
            "auth_port": null, 
            "encrypt_type": null, 
            "host": "nxos01", 
            "host_timeout": null, 
            "key": null, 
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "port": 80, 
            "provider": {
                "host": "nxos01", 
                "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                "port": 80, 
                "ssh_keyfile": null, 
                "timeout": 10, 
                "transport": "nxapi", 
                "use_ssl": false, 
                "username": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                "validate_certs": true
            }, 
            "server_type": "tacacs", 
            "ssh_keyfile": null, 
            "state": "absent", 
            "tacacs_port": null, 
            "timeout": 10, 
            "transport": "nxapi", 
            "url_password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "url_username": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "use_ssl": false, 
            "username": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "validate_certs": true
        }
    }, 
    "msg": "Pipe is not allowed for this messge type", 
    "output": {
        "clierror": "", 
        "code": "501", 
        "input": "show run | inc tacacs-server.host.8.8.8.8", 
        "msg": "Pipe is not allowed for this messge type"
    }, 
    "url": "http://nxos01:80/ins"
}

TASK [nxos_aaa_server_host : debug] **************************************************************************************************************************************************************************************************************
task path: /home/johnb/git/ansible-inc/nxos-prs/pr28939/test/integration/targets/nxos_aaa_server_host/tests/common/radius.yaml:2
ok: [nxos01] => {
    "msg": "START TRANSPORT:nxapi nxos_aaa_server_host radius.yaml sanity test"
}

TASK [nxos_aaa_server_host : Setup] **************************************************************************************************************************************************************************************************************
task path: /home/johnb/git/ansible-inc/nxos-prs/pr28939/test/integration/targets/nxos_aaa_server_host/tests/common/radius.yaml:4
<nxos01> connection transport is nxapi
Using module file /home/johnb/git/ansible-inc/ansible-workspace-4/lib/ansible/modules/network/nxos/nxos_aaa_server_host.py
<nxos01> ESTABLISH LOCAL CONNECTION FOR USER: johnb
<nxos01> EXEC /bin/sh -c 'echo ~ && sleep 0'
<nxos01> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/johnb/.ansible/tmp/ansible-tmp-1505310456.5-277613506983935 `" && echo ansible-tmp-1505310456.5-277613506983935="` echo /home/johnb/.ansible/tmp/ansible-tmp-1505310456.5-277613506983935 `" ) && sleep 0'
<nxos01> PUT /tmp/tmpJ9btoH TO /home/johnb/.ansible/tmp/ansible-tmp-1505310456.5-277613506983935/nxos_aaa_server_host.py
<nxos01> EXEC /bin/sh -c 'chmod u+x /home/johnb/.ansible/tmp/ansible-tmp-1505310456.5-277613506983935/ /home/johnb/.ansible/tmp/ansible-tmp-1505310456.5-277613506983935/nxos_aaa_server_host.py && sleep 0'
<nxos01> EXEC /bin/sh -c 'python /home/johnb/.ansible/tmp/ansible-tmp-1505310456.5-277613506983935/nxos_aaa_server_host.py; rm -rf "/home/johnb/.ansible/tmp/ansible-tmp-1505310456.5-277613506983935/" > /dev/null 2>&1 && sleep 0'
fatal: [nxos01]: FAILED! => {
    "changed": false, 
    "clierror": "", 
    "code": "501", 
    "failed": true, 
    "input": "show run | inc radius-server.host.8.8.8.8", 
    "invocation": {
        "module_args": {
            "acct_port": null, 
            "address": "8.8.8.8", 
            "auth_port": null, 
            "encrypt_type": null, 
            "host": "nxos01", 
            "host_timeout": null, 
            "key": null, 
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "port": 80, 
            "provider": {
                "host": "nxos01", 
                "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                "port": 80, 
                "ssh_keyfile": null, 
                "timeout": 10, 
                "transport": "nxapi", 
                "use_ssl": false, 
                "username": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                "validate_certs": true
            }, 
            "server_type": "radius", 
            "ssh_keyfile": null, 
            "state": "absent", 
            "tacacs_port": null, 
            "timeout": 10, 
            "transport": "nxapi", 
            "url_password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "url_username": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "use_ssl": false, 
            "username": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "validate_certs": true
        }
    }, 
    "msg": "Pipe is not allowed for this messge type", 
    "output": {
        "clierror": "", 
        "code": "501", 
        "input": "show run | inc radius-server.host.8.8.8.8", 
        "msg": "Pipe is not allowed for this messge type"
    }, 
    "url": "http://nxos01:80/ins"
}
...ignoring

TASK [nxos_aaa_server_host : Configure radius server defaults] ***********************************************************************************************************************************************************************************
task path: /home/johnb/git/ansible-inc/nxos-prs/pr28939/test/integration/targets/nxos_aaa_server_host/tests/common/radius.yaml:13
<nxos01> connection transport is nxapi
Using module file /home/johnb/git/ansible-inc/ansible-workspace-4/lib/ansible/modules/network/nxos/nxos_aaa_server_host.py
<nxos01> ESTABLISH LOCAL CONNECTION FOR USER: johnb
<nxos01> EXEC /bin/sh -c 'echo ~ && sleep 0'
<nxos01> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/johnb/.ansible/tmp/ansible-tmp-1505310457.68-110802324639560 `" && echo ansible-tmp-1505310457.68-110802324639560="` echo /home/johnb/.ansible/tmp/ansible-tmp-1505310457.68-110802324639560 `" ) && sleep 0'
<nxos01> PUT /tmp/tmp6SWkxz TO /home/johnb/.ansible/tmp/ansible-tmp-1505310457.68-110802324639560/nxos_aaa_server_host.py
<nxos01> EXEC /bin/sh -c 'chmod u+x /home/johnb/.ansible/tmp/ansible-tmp-1505310457.68-110802324639560/ /home/johnb/.ansible/tmp/ansible-tmp-1505310457.68-110802324639560/nxos_aaa_server_host.py && sleep 0'
<nxos01> EXEC /bin/sh -c 'python /home/johnb/.ansible/tmp/ansible-tmp-1505310457.68-110802324639560/nxos_aaa_server_host.py; rm -rf "/home/johnb/.ansible/tmp/ansible-tmp-1505310457.68-110802324639560/" > /dev/null 2>&1 && sleep 0'
fatal: [nxos01]: FAILED! => {
    "changed": false, 
    "clierror": "", 
    "code": "501", 
    "failed": true, 
    "input": "show run | inc radius-server.host.8.8.8.8", 
    "invocation": {
        "module_args": {
            "acct_port": null, 
            "address": "8.8.8.8", 
            "auth_port": null, 
            "encrypt_type": null, 
            "host": "nxos01", 
            "host_timeout": null, 
            "key": null, 
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "port": 80, 
            "provider": {
                "host": "nxos01", 
                "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                "port": 80, 
                "ssh_keyfile": null, 
                "timeout": 10, 
                "transport": "nxapi", 
                "use_ssl": false, 
                "username": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                "validate_certs": true
            }, 
            "server_type": "radius", 
            "ssh_keyfile": null, 
            "state": "present", 
            "tacacs_port": null, 
            "timeout": 10, 
            "transport": "nxapi", 
            "url_password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "url_username": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "use_ssl": false, 
            "username": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "validate_certs": true
        }
    }, 
    "msg": "Pipe is not allowed for this messge type", 
    "output": {
        "clierror": "", 
        "code": "501", 
        "input": "show run | inc radius-server.host.8.8.8.8", 
        "msg": "Pipe is not allowed for this messge type"
    }, 
    "url": "http://nxos01:80/ins"
}

TASK [nxos_aaa_server_host : debug] **************************************************************************************************************************************************************************************************************
task path: /home/johnb/git/ansible-inc/nxos-prs/pr28939/test/integration/targets/nxos_aaa_server_host/tests/common/radius.yaml:120
ok: [nxos01] => {
    "msg": "TRANSPORT:nxapi nxos_aaa_server_host failure detected"
}

```

**nxos_gir_profile_management/tests/common/sanity.yaml**
```
TASK [nxos_gir_profile_management : Setup - Turn on feature eigrp] *******************************************************************************************************************************************************************************
task path: /home/johnb/git/ansible-inc/nxos-prs/pr28939/test/integration/targets/nxos_gir_profile_management/tests/common/sanity.yaml:18
<nxos01> connection transport is cli
<nxos01> using connection plugin network_cli
<nxos01> socket_path: /home/johnb/.ansible/pc/7b6586159a
Using module file /home/johnb/git/ansible-inc/ansible-workspace-4/lib/ansible/modules/network/nxos/nxos_feature.py
<nxos01> ESTABLISH LOCAL CONNECTION FOR USER: johnb
<nxos01> EXEC /bin/sh -c 'echo ~ && sleep 0'
<nxos01> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/johnb/.ansible/tmp/ansible-tmp-1505310548.45-267845505082223 `" && echo ansible-tmp-1505310548.45-267845505082223="` echo /home/johnb/.ansible/tmp/ansible-tmp-1505310548.45-267845505082223 `" ) && sleep 0'
<nxos01> PUT /tmp/tmp8vKi8Y TO /home/johnb/.ansible/tmp/ansible-tmp-1505310548.45-267845505082223/nxos_feature.py
<nxos01> EXEC /bin/sh -c 'chmod u+x /home/johnb/.ansible/tmp/ansible-tmp-1505310548.45-267845505082223/ /home/johnb/.ansible/tmp/ansible-tmp-1505310548.45-267845505082223/nxos_feature.py && sleep 0'
<nxos01> EXEC /bin/sh -c 'python /home/johnb/.ansible/tmp/ansible-tmp-1505310548.45-267845505082223/nxos_feature.py; rm -rf "/home/johnb/.ansible/tmp/ansible-tmp-1505310548.45-267845505082223/" > /dev/null 2>&1 && sleep 0'
ok: [nxos01] => {
    "changed": false, 
    "commands": [], 
    "failed": false, 
    "invocation": {
        "module_args": {
            "config": null, 
            "feature": "eigrp", 
            "host": null, 
            "include_defaults": null, 
            "password": null, 
            "port": null, 
            "provider": {
                "host": null, 
                "password": null, 
                "port": null, 
                "ssh_keyfile": null, 
                "timeout": null, 
                "transport": "cli", 
                "use_ssl": null, 
                "username": null, 
                "validate_certs": null
            }, 
            "save": null, 
            "ssh_keyfile": null, 
            "state": "enabled", 
            "timeout": null, 
            "transport": "cli", 
            "use_ssl": null, 
            "username": null, 
            "validate_certs": null
        }
    }
}

TASK [nxos_gir_profile_management : Create maintenace mode profile] ******************************************************************************************************************************************************************************
task path: /home/johnb/git/ansible-inc/nxos-prs/pr28939/test/integration/targets/nxos_gir_profile_management/tests/common/sanity.yaml:26
<nxos01> connection transport is cli
<nxos01> using connection plugin network_cli
<nxos01> socket_path: /home/johnb/.ansible/pc/7b6586159a
Using module file /home/johnb/git/ansible-inc/ansible-workspace-4/lib/ansible/modules/network/nxos/nxos_gir_profile_management.py
<nxos01> ESTABLISH LOCAL CONNECTION FOR USER: johnb
<nxos01> EXEC /bin/sh -c 'echo ~ && sleep 0'
<nxos01> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/johnb/.ansible/tmp/ansible-tmp-1505310552.0-98170677845249 `" && echo ansible-tmp-1505310552.0-98170677845249="` echo /home/johnb/.ansible/tmp/ansible-tmp-1505310552.0-98170677845249 `" ) && sleep 0'
<nxos01> PUT /tmp/tmpZm0VtQ TO /home/johnb/.ansible/tmp/ansible-tmp-1505310552.0-98170677845249/nxos_gir_profile_management.py
<nxos01> EXEC /bin/sh -c 'chmod u+x /home/johnb/.ansible/tmp/ansible-tmp-1505310552.0-98170677845249/ /home/johnb/.ansible/tmp/ansible-tmp-1505310552.0-98170677845249/nxos_gir_profile_management.py && sleep 0'
<nxos01> EXEC /bin/sh -c 'python /home/johnb/.ansible/tmp/ansible-tmp-1505310552.0-98170677845249/nxos_gir_profile_management.py; rm -rf "/home/johnb/.ansible/tmp/ansible-tmp-1505310552.0-98170677845249/" > /dev/null 2>&1 && sleep 0'
fatal: [nxos01]: FAILED! => {
    "changed": false, 
    "failed": true, 
    "invocation": {
        "module_args": {
            "commands": [
                "router eigrp 11", 
                "isolate"
            ], 
            "config": null, 
            "host": null, 
            "include_defaults": "False", 
            "mode": "maintenance", 
            "password": null, 
            "port": null, 
            "provider": {
                "host": null, 
                "password": null, 
                "port": null, 
                "ssh_keyfile": null, 
                "timeout": null, 
                "transport": "cli", 
                "use_ssl": null, 
                "username": null, 
                "validate_certs": null
            }, 
            "ssh_keyfile": null, 
            "state": "present", 
            "timeout": null, 
            "transport": "cli", 
            "use_ssl": null, 
            "username": null, 
            "validate_certs": null
        }
    }, 
    "msg": "configure maintenance profile maintenance-mode\r\r\n                          ^\r\n% Invalid command at '^' marker.\r\n\rswitch(config)# "
}

TASK [nxos_gir_profile_management : debug] *******************************************************************************************************************************************************************************************************
task path: /home/johnb/git/ansible-inc/nxos-prs/pr28939/test/integration/targets/nxos_gir_profile_management/tests/common/sanity.yaml:92
ok: [nxos01] => {
    "msg": "TRANSPORT:cli nxos_gir_profile_management failure detected"
}

```

**Review by gundalow — CHANGES_REQUESTED** [review]
Needs revision as failing on our test equipment

**Comment by rahushen** [comment]
@gundalow ... the aaa_server_host issues should be resolved by a PR which has not yet been merged. The gir_profile_management issue also had a PR open which recently got merged. Can you verify that is failing after the PRs were merged ? Also, can you verify against 9K ? I don't have a titanium (7K VM) as we don't test against it as it is not a customer supported platform. If this fails only on titanium, please pass a diff to change the tests.

**Comment by gundalow** [comment]
@rahushen do you have links to the two PRs in question?

**Comment by rahushen** [comment]
@gundalow 

ansible/ansible#29103
ansible/ansible#29151

**Comment by gundalow** [comment]
@rahushen, thanks
As discussed in the Thursday 14th Sept call we will move the test cases to `Milestone:2.5.0`
Once 2.4.0 is done we will cherry pick this into `stable-2.4`, however this isn't blocking the release.


29151 is in `devel` and `stable-2.4`

**Comment by gundalow** [comment]
Rebase needed

**Comment by Qalthos** [comment]
Everything checks out except `nxos_gir_profile_management`, which tells me

```
switch# configure maintenance profile
                  ^
% Invalid command at '^' marker.
```

`configure profile maintenance-mode type admin` does seem to work, and yells at me to use the correct command. This seems like an NXOSv issue to me, especially if you have this working on a 7.3(0)D1(1)

**Comment by rahushen** [comment]
@Qalthos ... I just checked and it is definitely an NX-OSv only issue. I've updated the test to skip for the virtual platform. Please give it a try and merge if the tests pass.

**Review by gundalow — APPROVED** [review]

File: test/integration/targets/nxos_gir/tests/common/sanity.yaml
Comment (trishnaguha) [reply test/integration/targets/nxos_gir/tests/common/sanity.yaml:4]: @rahushen do you have any issue opened for the module?

File: test/integration/targets/nxos_aaa_server_host/tests/common/radius.yaml
Comment (trishnaguha) [inline test/integration/targets/nxos_aaa_server_host/tests/common/radius.yaml]: @rahushen Idempotence test failed here for both transports on NXOS 7K and 9K.

File: test/integration/targets/nxos_aaa_server_host/tests/common/tacacs.yaml
Comment (trishnaguha) [inline test/integration/targets/nxos_aaa_server_host/tests/common/tacacs.yaml]: @rahushen Idempotence test failed here for both transports on nxos 7K and 9K.

**Review by trishnaguha — DISMISSED** [review]
The module needs to be fixed before we merge the PR.

**Comment by rahushen** [comment]
@trishnaguha ... That is weird. I can understand the 7K vm failing, but the 9K VMs are passing the tests in our environment. Did you run the tests against devel ?

File: test/integration/targets/nxos_gir/tests/common/sanity.yaml
Comment (rahushen) [reply test/integration/targets/nxos_gir/tests/common/sanity.yaml:4]: @trishnaguha ... I had opened #27292 for this. I tried setting the connection timeout as you mentioned and it seems to work, but the module has other issues.

**Comment by trishnaguha** [comment]
@rahushen yeah devel. I can see why.
before applying the task:
```
 0.6s main: 
      existing={}
 0.6s main: 
      proposed={'acct_port': '2084', 'address': '8.8.8.8', 'auth_port': '2083', 'encrypt_type': '0', 'host_timeout': '25', 'key': 'hello', 'server_type': 'radius'}
```
idempotence run:
```
 0.6s main: 
      existing={'acct_port': u'2084', 'address': '8.8.8.8', 'auth_port': u'2083', 'host_timeout': u'25', u'key': u'kapqg', 'server_type': 'radius'}
 0.6s main: 
      proposed={'acct_port': '2084', 'address': '8.8.8.8', 'auth_port': '2083', 'encrypt_type': '0', 'host_timeout': '25', 'key': 'hello', 'server_type': 'radius'}
 0.6s main: delta={'encrypt_type': '0', 'key': 'hello'}
```

File: test/integration/targets/nxos_aaa_server_host/tests/common/radius.yaml
Comment (trishnaguha) [inline test/integration/targets/nxos_aaa_server_host/tests/common/radius.yaml:77]: @rahushen after testing `nxos_aaa_server_host` on nxos7K and 9K, it seems that the hosts force the encrypt_type to 7 even providing 0 as encrypt_type, hence the idempotence issue. will you restrict some of these testcases to other versions of NXOS other than 7K and 9K? using `when: platform is search('N5K')`?

Or is there anyway to make the hosts not forcing the encrypt_type?

File: test/integration/targets/nxos_aaa_server_host/tests/common/tacacs.yaml
Comment (trishnaguha) [inline test/integration/targets/nxos_aaa_server_host/tests/common/tacacs.yaml:82]: Same like above https://github.com/ansible/ansible/pull/28939/files#r156870128

**Comment by rahushen** [comment]
@trishnaguha ... I think I've solved the mystery regarding why these tests were passing in our setup. Apparently, idempotency is not supported for clear text passwords and I made this change to out local testing repository. It unfortunately never made this PR and hence you were seeing the issue. I've now synced the nxos_aaa_server_host tests with the local repository we maintain and pushed it out to this PR. Please give it a try and merge if tests pass for you.

**Review by trishnaguha — APPROVED** [review]

**Comment by trishnaguha** [comment]
cherry-picked to 2.4

**Pull request merged**
